tools/distrib/check_include_guards.py

```diff
@@ -95,56 +95,59 @@ def check(self, fpath, fix):
     fcontents = load(fpath)
 
     match = self.ifndef_re.search(fcontents)
-    if match.lastindex != 1:
+    if match.lastindex is None:
       # No ifndef. Request manual addition with hints
       self.fail(fpath, match.re, match.string, '', '', False)
+      return False  # failed
 
     # Does the guard end with a '_H'?
     running_guard = match.group(1)
     if not running_guard.endswith('_H'):
       fcontents = self.fail(fpath, match.re, match.string, match.group(1),
                             valid_guard, fix)
-      save(fpath, fcontents)
+      if fix: save(fpath, fcontents)
 
     # Is it the expected one based on the file path?
     if running_guard != valid_guard:
       fcontents = self.fail(fpath, match.re, match.string, match.group(1),
                             valid_guard, fix)
-      save(fpath, fcontents)
+      if fix: save(fpath, fcontents)
 
     # Is there a #define? Is it the same as the #ifndef one?
     match = self.define_re.search(fcontents)
-    if match.lastindex != 1:
+    if match.lastindex is None:
       # No define. Request manual addition with hints
       self.fail(fpath, match.re, match.string, '', '', False)
+      return False  # failed
 
     # Is the #define guard the same as the #ifndef guard?
     if match.group(1) != running_guard:
       fcontents = self.fail(fpath, match.re, match.string, match.group(1),
                             valid_guard, fix)
-      save(fpath, fcontents)
+      if fix: save(fpath, fcontents)
 
     # Is there a properly commented #endif?
     endif_re = self.endif_cpp_re if cpp_header else self.endif_c_re
-    matches = endif_re.findall(fcontents)
-    if not matches:
+    flines = fcontents.rstrip().splitlines()
```

Comment:
I am learning a new thing every day!  - in this case the `rstrip()` function .. thanks! :)